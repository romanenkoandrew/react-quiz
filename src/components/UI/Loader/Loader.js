import React from "react";
import styles from "./Loader.module.css";

const Loader = () => (
  <div className={styles.ldsRoller}>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
);

export default Loader;
