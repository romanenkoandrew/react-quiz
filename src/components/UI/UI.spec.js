import React from "react";
import { shallow } from "enzyme";
import Backdrop from "./Backdrop/Backdrop";
import Button from "./Button/Button";
import Loader from "./Loader/Loader";
import Select from "./Select/Select";
import Input from "./Input/Input";

describe("Backdrop component", () => {
  it("should render Backdrop with props.isOpen", () => {
    const component = shallow(<Backdrop />);
    expect(component).toMatchSnapshot();
  });
});
describe("Button component", () => {
  it("should render Button", () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });
});
describe("Loader component", () => {
  it("should render Loader", () => {
    const component = shallow(<Loader />);
    expect(component).toMatchSnapshot();
  });
});
describe("Select component", () => {
  it("should render Select", () => {
    const component = shallow(
      <Select options={[{ value: "value ", text: "text" }]} />
    );
    expect(component).toMatchSnapshot();
  });
});
describe("Input component", () => {
  it("should render default Input", () => {
    const component = shallow(
      <Input
        type={"type"}
        value={"value"}
        errorMessage={"errorMessage "}
        valid
        touched
        shouldValidate
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("should render Input without props.type", () => {
    const component = shallow(
      <Input value={"value"} valid={false} touched shouldValidate />
    );
    expect(component).toMatchSnapshot();
  });
});
