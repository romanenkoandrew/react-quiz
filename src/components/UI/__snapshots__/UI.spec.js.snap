// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop component should render Backdrop with props.isOpen 1`] = `
<div
  className="Backdrop"
/>
`;

exports[`Button component should render Button 1`] = `
<button
  className="Button undefined"
/>
`;

exports[`Input component should render Input without props.type 1`] = `
<div
  className="Input invalid"
>
  <label
    htmlFor="text--0.21030961908189472"
  />
  <input
    id="text--0.21030961908189472"
    type="text"
    value="value"
  />
  <span>
    Введите верное значение
  </span>
</div>
`;

exports[`Input component should render default Input 1`] = `
<div
  className="Input"
>
  <label
    htmlFor="type--0.7813774796250152"
  />
  <input
    id="type--0.7813774796250152"
    type="type"
    value="value"
  />
</div>
`;

exports[`Loader component should render Loader 1`] = `
<div
  className="ldsRoller"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Select component should render Select 1`] = `
<div
  className="Select"
>
  <label
    htmlFor="undefined-0.05517909705267399"
  />
  <select
    id="undefined-0.05517909705267399"
  >
    <option
      key="value 0"
      value="value "
    >
      text
    </option>
  </select>
</div>
`;
